ote/src/clj/ote/palvelut/transport.clj

```diff
@@ -88,11 +102,40 @@
                   (update ::t-service/passenger-transportation dissoc ::t-service/operation_area)
                   (update-in [::t-service/passenger-transportation ::t-service/price-classes] fix-price-classes))]
     (jdbc/with-db-transaction [db db]
-      (let [transport-service
-            (upsert! db ::t-service/transport-service value)]
+         (let [new-value (dissoc value :transport-service)
+               transport-service (upsert! db ::t-service/transport-service new-value)]
         (places/link-places-to-transport-service!
          db (::t-service/id transport-service) places)))))
 
+(defn- save-terminal-info
+  "UPSERT! given data to database. "
+  [db data]
+  (println "Terminal DATA: " (pr-str data))
```

Comment:
Add access controls here. User must be associated to the transport-operator they are trying to add a service for.

Reply:
Created new ticket for this